rundeckapp/grails-app/services/rundeck/services/RdJobService.groovy

```diff
@@ -0,0 +1,54 @@
+package rundeck.services
+
+import groovy.transform.CompileStatic
+import org.rundeck.app.data.model.v1.DeletionResult
+import org.rundeck.app.data.model.v1.job.JobData
+import org.rundeck.app.data.model.v1.query.JobQueryInputData
+import org.rundeck.app.data.providers.v1.job.JobDataProvider
+
+@CompileStatic
+class RdJobService {
+
+    JobDataProvider jobDataProvider
+
+    JobData getJobByIdOrUuid(Serializable id) {
+        JobData found = null
+        if (id instanceof Long) {
+            return getJobById(id)
+        } else if (id instanceof String) {
+            //attempt to parse as long id
+            try {
+                def idlong = Long.parseLong(id)
+                found = getJobById(idlong)
+            } catch (NumberFormatException e) {
```

Comment:
Should we at least log.info these in case an exception is thrown?

Reply:
In this case it's fine to ignore this error. In a future PR I have marked this method of returning the job as deprecated because we should only be using the uuid in the future and not the internal.